### DISCUSSION TITLE
spdlog backtrace

### DISCUSSION THREAD

**Discussion by samuelnaga** [gabime/spdlog#1965] [Q&A]
Hello,
I wondered if there is any way to "stop" logging while performing some operations that require high performance.
The idea would be to save the logs (within these operations) in memory and, when the operations have finished, the saved logs get written in the log file as normally. 
This is more or less what I need:

    auto logger = spdlog::basic_logger_mt("basic_logger", "basic-log.txt");
    logger->set_level(spdlog::level::debug);
    logger->info("Initial LOG");
    logger->info("Starting operations...");
    logger->enable_backtrace(20);
    for(int i = 0; i < 20; i++)
        logger->debug("Operation {}", i); // this log should not be written until dump_backtrace() is called

    logger->dump_backtrace();
    logger->disable_backtrace();

However, this code is writing the debug log from the operations twice, at operations runtime and again when dumb_backtrace() is called. 
Is there a way to just write this log just when calling dump_backtrace() or similar function?

I have noticed that when a higher log level than the defined one (trace in this case) is used in the operations loop,  they are just written when the dumb_backtrace() is called. That would be the desired behavior but, regardless  the log level.

**Comment by gabime** [comment]
> I have noticed that when a higher log level than the defined one (trace in this case) is used in the operations loop, they are just written when the dumb_backtrace() is called. That would be the desired behavior but, regardless the log level.

Right, this is by design. Just set to high level to turn it off while still keeping the backtrace.
